src/main/java/dev/openfeature/sdk/Structure.java

```diff
@@ -46,6 +46,14 @@ public interface Structure {
      */
     Map<String, Value> asMap();
 
+    /**
+     * Get all values, as a map of Values.
+     *
+     * @return all attributes on the structure into a Map
+     */
+    Map<String, Value> asUnmodifiableMap();
```

Comment:
We can make good use of this internally when we want to just get the attributes as a map but don't care if we actually copy them or not - it just returns an immutable view, not a copy.